aws/resource_aws_ssm_maintenance_window.go

```diff
@@ -143,6 +149,12 @@ func resourceAwsSsmMaintenanceWindowUpdate(d *schema.ResourceData, meta interfac
 		params.StartDate = aws.String(v.(string))
 	}
 
+	if d.HasChange("tags") {
+		if err := setTagsSSM(ssmconn, d, d.Id(), ssm.ResourceTypeForTaggingMaintenanceWindow); err != nil {
+			return fmt.Errorf("error setting SSM Maintenance Window tags: %s", err)
+		}
+	}
```

Comment:
It might be safer to move this code below the logic which actually updates the maintenance window as that already has some logic for handling 404 (e.g. when the resource is manually nuked and Terraform doesn't do refresh prior to update, for any reason).